### PR TITLE
Fix utf8 range check in SCC

### PR DESCRIPTION
**Pull request by tajila**
Fix utf8 range check in SCC

The existing check dereferences the utf8 before ensuring that the header
is in the SCC. This causes a segfault when the header is not in the SCC.

Signed-off-by: Tobi Ajila <atobia@ca.ibm.com>

### CODE REVIEW

**Comment by tajila** [comment]
@hangshao0 please review

**Comment by tajila** [comment]
FYI @gacholio

File: runtime/bcutil/ComparingCursor.cpp [hunk 1]
```diff
@@ -541,7 +541,8 @@ ComparingCursor::isRangeValidForUTF8Ptr(J9UTF8 *utf8)
 	U_8 *ptr = (U_8*)utf8;
 
 	if (_checkRangeInSharedCache) {
-		return FALSE != j9shr_Query_IsAddressInCache(_javaVM, utf8, J9UTF8_TOTAL_SIZE(utf8));
+		return FALSE != (j9shr_Query_IsAddressInCache(_javaVM, utf8, sizeof(J9UTF8))
```
Comment (pshipton) [inline runtime/bcutil/ComparingCursor.cpp:544]: Pls remove the redundant `FALSE !=`

**Comment by pshipton** [comment]
A little more explanation would be helpful for why a utf8 that can't be deferenced is being checked at all. Please add a comment to the code to explain why the two checks are required.

File: runtime/bcutil/ComparingCursor.cpp (hunk 1 above)
Comment (pshipton) [reply runtime/bcutil/ComparingCursor.cpp:544]: and the redundant brackets.

Comment (pshipton) [reply runtime/bcutil/ComparingCursor.cpp:544]: Hmm, nm I guess this is needed to convert a C boolean into a C++ boolean.

Comment (tajila) [reply runtime/bcutil/ComparingCursor.cpp]: I think with the `&&` we should be fine. Ill try it on windows to verify.

**Review by hangshao0 — COMMENTED** [review]
LGTM.

**Review by hangshao0 — APPROVED** [review]

**Comment by pshipton** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
@pshipton I've update the comments. Please let me know if it addresses your concerns.

The windows build passed

**Review by pshipton — APPROVED** [review]

**Pull request merged**